tests/test_time.py

```diff
@@ -5,20 +5,129 @@
 import pytest
 import xarray as xr
 from s2spy.time import AdventCalendar
+from s2spy.time import CustomCalendar
 from s2spy.time import MonthlyCalendar
+from s2spy.time import Period
+from s2spy.time import PrecursorPeriod
+from s2spy.time import TargetPeriod
 from s2spy.time import WeeklyCalendar
 
 
-def interval(start, end):
+def interval(start, end, closed="right"):
     """Shorthand for more readable tests."""
-    return pd.Interval(pd.Timestamp(start), pd.Timestamp(end))
+    return pd.Interval(pd.Timestamp(start), pd.Timestamp(end), closed = closed)
+
+
+class TestPeriod:
+    """Test Period objects."""
+
+    def test_period(self):
+        period = Period(length=10, gap=10)
+        assert isinstance(period, Period)
+
+    def test_target_period(self):
+        target = TargetPeriod(20, 10)
+        assert isinstance(target, TargetPeriod)
+
+    def test_precursor_period(self):
+        precursor = PrecursorPeriod(10, -5)
+        assert isinstance(precursor, PrecursorPeriod)
+
+
+class TestCustomCalendar:
+    """Test CustomCalendar methods."""
+
+    @pytest.fixture(autouse=True)
+    def dummy_calendar(self):
+        cal = CustomCalendar(anchor="12-31")
+        # create target periods
+        target_1 = TargetPeriod(20)
+        # create precursor periods
+        precursor_1 = PrecursorPeriod(10)
+        # append building blocks
+        cal.append(target_1)
+        cal.append(precursor_1)
+        # map years
+        cal = cal.map_years(2021, 2021)
```

Comment:
The map_to_data method is not tested here yet

Reply:
Add a test for `map_to_data` now.